### DISCUSSION TITLE
How to use so-vits-svc-fork in discord?

### DISCUSSION THREAD

**Discussion by sdfstudios** [voicepaw/so-vits-svc-fork#500] [Q&A] [answered]
i want to use so-vits-svc-fork in discord. but I don't understand how to do it. the program has a real-time mode, but how to connect it to the discord?

**Comment by athahibatullah** [comment, marked as answer]
I use an additional program named **Voicemeeter** to do that. Here I'll just explain my own setting, it may vary depending on your device. I assume you are using **Windows OS**.
You'll need these additional program. No need to pay since they are free
- Virtual Audio Cable
- Voicemeeter Banana
- EarTrumpet (It's additional, you can get it from microsoft store)

Install them and open the Voicemeeter program.
Next, follow these steps:
**WARNING! These steps might cause your audio missing. No need to panic, you can just Set as Default Device your main audio to revert the changes or just screenshoot all your setting before doing these steps.**
1. Go to **Control Panel > Hardware and Sound > Sound**
![image](https://user-images.githubusercontent.com/31439960/234969753-505b06e7-6e25-444d-a8c6-50e84ba91043.png)

2. On the playback tab, scroll down and find **VoiceMeeter Input**. After that right click and **Set as Default Device**
![image](https://user-images.githubusercontent.com/31439960/234970941-b4806614-8b56-489f-a911-a11696a8a7a3.png)

3. Still on the playback tab, choose your mic and **Set as Default Communications Device** 
![image](https://user-images.githubusercontent.com/31439960/234972030-beff4c96-502c-49d2-90a9-5c5d4ce8a58b.png)

4. Move on to Recording tab, find **Cable Output** and **Set as Default Communications Device**
![image](https://user-images.githubusercontent.com/31439960/234973466-ba8f7179-a0e5-48de-93ca-6fdf9cdd5619.png)
 
5. Find **VoiceMeeter Output** and **Set as Default Device**
![image](https://user-images.githubusercontent.com/31439960/234974012-f3031c26-c8f9-4dfe-86c3-18dc2554fff9.png)
 
6. Open Voicemeeter and set your output device (your mic)
![image](https://user-images.githubusercontent.com/31439960/234976144-2e4e4de7-29d5-4c50-87db-09ef8b722047.png)
![image](https://user-images.githubusercontent.com/31439960/234976339-365af73d-5c59-4277-9c58-75433394daaf.png)

7. Set your input device by choosing Cable Output
![image](https://user-images.githubusercontent.com/31439960/234976727-2bde97e9-1d88-4d2f-ba0e-3f097533e0c0.png)

8. Set your input device like this
![image](https://user-images.githubusercontent.com/31439960/234978652-e8b17690-cae5-4247-967b-af6572c784d9.png)

And your final setting should look like this
![image](https://user-images.githubusercontent.com/31439960/234978939-747830bd-783f-460c-bd24-b77a167e6327.png)

9. Open your so-vits program and set **your mic** as input device and **Cable Input** as output device, and start the voice changer
![image](https://user-images.githubusercontent.com/31439960/234977789-b306f8f8-ca0b-422a-a17c-b02e634289d5.png)

10. If you want to know whether your so-vits are ready to connect with discord. When you speak using voice changer, this bar should raise
![image](https://user-images.githubusercontent.com/31439960/234979712-30f422a6-38d8-4c63-9125-f3d0b917a32f.png)

11. Open your discord voice setting and set **Voicemeeter** as both input device and output device
![image](https://user-images.githubusercontent.com/31439960/234978151-175c3895-0bb8-4b99-a23a-ee9968467437.png)

Check your voice using mic test on discord to know if you already connect your so-vits to your discord.

12. Additional steps. If you can still hear your converted voice and want to mute your voice, use EarTrumpet and mute the cable output
![image](https://user-images.githubusercontent.com/31439960/234980643-cdc9bda7-1c40-479b-8f5a-db022467ce74.png)

If you want to revert everything back, just **disable** all of the settings and **set default** your main audio on  **Control Panel > Hardware and Sound > Sound**

The main idea to integrate so-vits with discord:
Your main mic > voice change > cable input (use output from so-vits as input on voicemeeter) >  cable output (receiving the output from so-vits as input on voicemeeter) > virtual out (use output from voicemeeter as input on discord) > voicemeeter output (receiving the output from voicemeeter as input on discord) > Your changed voice appear on discord

That's all I can share with you. I'm new with this kind of audio setup. If you have any issues other than I've faced then you should start exploring on yourself.

**Reply by ADR3-N** [reply]
I should add that Voice Meeter is an optional quality of life improvement. I've been using so-vits-svc realtime function to record music and only ever needed Virtual Audio Cable and manually shifting audio levels via hardware or mixer panel.

Just make sure you route your microphone to so-vits, and so-vits output to virtual audio input, THEN set Discord to recognize virtual audio OUTPUT as your mic. There should be no problems.
